src/Simplex/Chat/Protocol.hs

```diff
@@ -243,6 +248,13 @@ instance ToJSON SharedMsgId where
   toJSON = strToJSON
   toEncoding = strToJEncoding
 
+data MsgScope
+  = MSGroup
+  | MSMember {memberId :: MemberId} -- Admins can use any member id; members can use only their own id
+  deriving (Eq, Show)
+
+$(JQ.deriveJSON (sumTypeJSON $ dropPrefix "MS") ''MsgScope)
```

Comment:
you can't use platform-dependent encoding in protocol... tsk tsk.

```suggestion
$(JQ.deriveJSON (taggedObjectJSON $ dropPrefix "MS") ''MsgScope)
```

Reply:
which means that the instance has to be manually derived in swift

Reply:
this type is not needed in swift